training_runs/0_training_0/dynapse_parameters.txt
C0c0:(IF_AHTAU_N,4,80)
C0c0:(IF_AHTHR_N,4,80)
C0c0:(IF_AHW_P,0,0)
C0c0:(IF_BUF_P,4,80)
C0c0:(IF_CASC_N,0,0)
C0c0:(IF_DC_P,0,0)
C0c0:(IF_NMDA_N,0,0)
C0c0:(IF_RFR_N,2,128)
C0c0:(IF_TAU1_N,3,150)
C0c0:(IF_TAU2_N,3,150)
C0c0:(IF_THR_N,0,0)
C0c0:(NPDPIE_TAU_F_P,7,255)
C0c0:(NPDPIE_TAU_S_P,2,100)
C0c0:(NPDPIE_THR_F_P,5,200)
C0c0:(NPDPIE_THR_S_P,5,200)
C0c0:(NPDPII_TAU_F_P,7,255)
C0c0:(NPDPII_TAU_S_P,7,255)
C0c0:(NPDPII_THR_F_P,5,200)
C0c0:(NPDPII_THR_S_P,5,200)
C0c0:(PS_WEIGHT_EXC_F_N,6,200)
C0c0:(PS_WEIGHT_EXC_S_N,7,130)
C0c0:(PS_WEIGHT_INH_F_N,6,120)
C0c0:(PS_WEIGHT_INH_S_N,7,255)
C0c0:(PULSE_PWLK_P,4,106)
C0c0:(R2R_P,3,85)
C0c1:(IF_AHTAU_N,4,80)
C0c1:(IF_AHTHR_N,4,80)
C0c1:(IF_AHW_P,0,0)
C0c1:(IF_BUF_P,4,80)
C0c1:(IF_CASC_N,0,0)
C0c1:(IF_DC_P,0,0)
C0c1:(IF_NMDA_N,0,0)
C0c1:(IF_RFR_N,2,128)
C0c1:(IF_TAU1_N,3,150)
C0c1:(IF_TAU2_N,3,150)
C0c1:(IF_THR_N,0,0)
C0c1:(NPDPIE_TAU_F_P,7,255)
C0c1:(NPDPIE_TAU_S_P,2,100)
C0c1:(NPDPIE_THR_F_P,5,200)
C0c1:(NPDPIE_THR_S_P,5,200)
C0c1:(NPDPII_TAU_F_P,7,255)
C0c1:(NPDPII_TAU_S_P,7,255)
C0c1:(NPDPII_THR_F_P,5,200)
C0c1:(NPDPII_THR_S_P,5,200)
C0c1:(PS_WEIGHT_EXC_F_N,6,200)
C0c1:(PS_WEIGHT_EXC_S_N,7,130)
C0c1:(PS_WEIGHT_INH_F_N,6,120)
C0c1:(PS_WEIGHT_INH_S_N,7,255)
C0c1:(PULSE_PWLK_P,4,106)
C0c1:(R2R_P,3,85)
C0c2:(IF_AHTAU_N,4,80)
C0c2:(IF_AHTHR_N,4,80)
C0c2:(IF_AHW_P,0,0)
C0c2:(IF_BUF_P,4,80)
C0c2:(IF_CASC_N,0,0)
C0c2:(IF_DC_P,0,0)
C0c2:(IF_NMDA_N,0,0)
C0c2:(IF_RFR_N,2,128)
C0c2:(IF_TAU1_N,3,150)
C0c2:(IF_TAU2_N,3,150)
C0c2:(IF_THR_N,0,0)
C0c2:(NPDPIE_TAU_F_P,7,255)
C0c2:(NPDPIE_TAU_S_P,2,100)
C0c2:(NPDPIE_THR_F_P,5,200)
C0c2:(NPDPIE_THR_S_P,5,200)
C0c2:(NPDPII_TAU_F_P,3,150)
C0c2:(NPDPII_TAU_S_P,3,150)
C0c2:(NPDPII_THR_F_P,5,200)
C0c2:(NPDPII_THR_S_P,5,200)
C0c2:(PS_WEIGHT_EXC_F_N,6,200)
C0c2:(PS_WEIGHT_EXC_S_N,4,50)
C0c2:(PS_WEIGHT_INH_F_N,6,120)
C0c2:(PS_WEIGHT_INH_S_N,7,255)
C0c2:(PULSE_PWLK_P,4,106)
C0c2:(R2R_P,3,85)
C0c3:(IF_AHTAU_N,4,80)
C0c3:(IF_AHTHR_N,4,80)
C0c3:(IF_AHW_P,0,0)
C0c3:(IF_BUF_P,4,80)
C0c3:(IF_CASC_N,0,0)
C0c3:(IF_DC_P,0,0)
C0c3:(IF_NMDA_N,0,0)
C0c3:(IF_RFR_N,2,128)
C0c3:(IF_TAU1_N,3,150)
C0c3:(IF_TAU2_N,3,150)
C0c3:(IF_THR_N,0,0)
C0c3:(NPDPIE_TAU_F_P,7,255)
C0c3:(NPDPIE_TAU_S_P,2,100)
C0c3:(NPDPIE_THR_F_P,5,200)
C0c3:(NPDPIE_THR_S_P,5,200)
C0c3:(NPDPII_TAU_F_P,3,150)
C0c3:(NPDPII_TAU_S_P,3,150)
C0c3:(NPDPII_THR_F_P,5,200)
C0c3:(NPDPII_THR_S_P,5,200)
C0c3:(PS_WEIGHT_EXC_F_N,6,200)
C0c3:(PS_WEIGHT_EXC_S_N,4,50)
C0c3:(PS_WEIGHT_INH_F_N,6,120)
C0c3:(PS_WEIGHT_INH_S_N,7,255)
C0c3:(PULSE_PWLK_P,4,106)
C0c3:(R2R_P,3,85)
C1c0:(IF_AHTAU_N,4,80)
C1c0:(IF_AHTHR_N,4,80)
C1c0:(IF_AHW_P,0,0)
C1c0:(IF_BUF_P,4,80)
C1c0:(IF_CASC_N,0,0)
C1c0:(IF_DC_P,0,0)
C1c0:(IF_NMDA_N,0,0)
C1c0:(IF_RFR_N,2,128)
C1c0:(IF_TAU1_N,3,150)
C1c0:(IF_TAU2_N,3,150)
C1c0:(IF_THR_N,0,0)
C1c0:(NPDPIE_TAU_F_P,7,255)
C1c0:(NPDPIE_TAU_S_P,2,100)
C1c0:(NPDPIE_THR_F_P,5,200)
C1c0:(NPDPIE_THR_S_P,5,200)
C1c0:(NPDPII_TAU_F_P,7,255)
C1c0:(NPDPII_TAU_S_P,7,255)
C1c0:(NPDPII_THR_F_P,5,200)
C1c0:(NPDPII_THR_S_P,5,200)
C1c0:(PS_WEIGHT_EXC_F_N,6,200)
C1c0:(PS_WEIGHT_EXC_S_N,7,130)
C1c0:(PS_WEIGHT_INH_F_N,6,120)
C1c0:(PS_WEIGHT_INH_S_N,7,255)
C1c0:(PULSE_PWLK_P,4,106)
C1c0:(R2R_P,3,85)
C1c1:(IF_AHTAU_N,4,80)
C1c1:(IF_AHTHR_N,4,80)
C1c1:(IF_AHW_P,0,0)
C1c1:(IF_BUF_P,4,80)
C1c1:(IF_CASC_N,0,0)
C1c1:(IF_DC_P,0,0)
C1c1:(IF_NMDA_N,0,0)
C1c1:(IF_RFR_N,2,128)
C1c1:(IF_TAU1_N,3,150)
C1c1:(IF_TAU2_N,3,150)
C1c1:(IF_THR_N,0,0)
C1c1:(NPDPIE_TAU_F_P,7,255)
C1c1:(NPDPIE_TAU_S_P,2,100)
C1c1:(NPDPIE_THR_F_P,5,200)
C1c1:(NPDPIE_THR_S_P,5,200)
C1c1:(NPDPII_TAU_F_P,7,255)
C1c1:(NPDPII_TAU_S_P,7,255)
C1c1:(NPDPII_THR_F_P,5,200)
C1c1:(NPDPII_THR_S_P,5,200)
C1c1:(PS_WEIGHT_EXC_F_N,6,200)
C1c1:(PS_WEIGHT_EXC_S_N,7,130)
C1c1:(PS_WEIGHT_INH_F_N,6,120)
C1c1:(PS_WEIGHT_INH_S_N,7,255)
C1c1:(PULSE_PWLK_P,4,106)
C1c1:(R2R_P,3,85)
C1c2:(IF_AHTAU_N,4,80)
C1c2:(IF_AHTHR_N,4,80)
C1c2:(IF_AHW_P,0,0)
C1c2:(IF_BUF_P,4,80)
C1c2:(IF_CASC_N,0,0)
C1c2:(IF_DC_P,0,0)
C1c2:(IF_NMDA_N,0,0)
C1c2:(IF_RFR_N,2,128)
C1c2:(IF_TAU1_N,3,150)
C1c2:(IF_TAU2_N,3,150)
C1c2:(IF_THR_N,0,0)
C1c2:(NPDPIE_TAU_F_P,7,255)
C1c2:(NPDPIE_TAU_S_P,2,100)
C1c2:(NPDPIE_THR_F_P,5,200)
C1c2:(NPDPIE_THR_S_P,5,200)
C1c2:(NPDPII_TAU_F_P,3,150)
C1c2:(NPDPII_TAU_S_P,3,150)
C1c2:(NPDPII_THR_F_P,5,200)
C1c2:(NPDPII_THR_S_P,5,200)
C1c2:(PS_WEIGHT_EXC_F_N,6,200)
C1c2:(PS_WEIGHT_EXC_S_N,4,50)
C1c2:(PS_WEIGHT_INH_F_N,6,120)
C1c2:(PS_WEIGHT_INH_S_N,7,255)
C1c2:(PULSE_PWLK_P,4,106)
C1c2:(R2R_P,3,85)
C1c3:(IF_AHTAU_N,4,80)
C1c3:(IF_AHTHR_N,4,80)
C1c3:(IF_AHW_P,0,0)
C1c3:(IF_BUF_P,4,80)
C1c3:(IF_CASC_N,0,0)
C1c3:(IF_DC_P,0,0)
C1c3:(IF_NMDA_N,0,0)
C1c3:(IF_RFR_N,2,128)
C1c3:(IF_TAU1_N,3,150)
C1c3:(IF_TAU2_N,3,150)
C1c3:(IF_THR_N,0,0)
C1c3:(NPDPIE_TAU_F_P,7,255)
C1c3:(NPDPIE_TAU_S_P,2,100)
C1c3:(NPDPIE_THR_F_P,5,200)
C1c3:(NPDPIE_THR_S_P,5,200)
C1c3:(NPDPII_TAU_F_P,3,150)
C1c3:(NPDPII_TAU_S_P,3,150)
C1c3:(NPDPII_THR_F_P,5,200)
C1c3:(NPDPII_THR_S_P,5,200)
C1c3:(PS_WEIGHT_EXC_F_N,6,200)
C1c3:(PS_WEIGHT_EXC_S_N,4,50)
C1c3:(PS_WEIGHT_INH_F_N,6,120)
C1c3:(PS_WEIGHT_INH_S_N,7,255)
C1c3:(PULSE_PWLK_P,4,106)
C1c3:(R2R_P,3,85)
C2c0:(IF_AHTAU_N,4,80)
C2c0:(IF_AHTHR_N,4,80)
C2c0:(IF_AHW_P,0,0)
C2c0:(IF_BUF_P,4,80)
C2c0:(IF_CASC_N,0,0)
C2c0:(IF_DC_P,0,0)
C2c0:(IF_NMDA_N,0,0)
C2c0:(IF_RFR_N,2,128)
C2c0:(IF_TAU1_N,3,150)
C2c0:(IF_TAU2_N,3,150)
C2c0:(IF_THR_N,0,0)
C2c0:(NPDPIE_TAU_F_P,7,255)
C2c0:(NPDPIE_TAU_S_P,2,100)
C2c0:(NPDPIE_THR_F_P,5,200)
C2c0:(NPDPIE_THR_S_P,5,200)
C2c0:(NPDPII_TAU_F_P,7,255)
C2c0:(NPDPII_TAU_S_P,7,255)
C2c0:(NPDPII_THR_F_P,5,200)
C2c0:(NPDPII_THR_S_P,5,200)
C2c0:(PS_WEIGHT_EXC_F_N,6,200)
C2c0:(PS_WEIGHT_EXC_S_N,7,130)
C2c0:(PS_WEIGHT_INH_F_N,6,120)
C2c0:(PS_WEIGHT_INH_S_N,7,255)
C2c0:(PULSE_PWLK_P,4,106)
C2c0:(R2R_P,3,85)
C2c1:(IF_AHTAU_N,4,80)
C2c1:(IF_AHTHR_N,4,80)
C2c1:(IF_AHW_P,0,0)
C2c1:(IF_BUF_P,4,80)
C2c1:(IF_CASC_N,0,0)
C2c1:(IF_DC_P,0,0)
C2c1:(IF_NMDA_N,0,0)
C2c1:(IF_RFR_N,2,128)
C2c1:(IF_TAU1_N,3,150)
C2c1:(IF_TAU2_N,3,150)
C2c1:(IF_THR_N,0,0)
C2c1:(NPDPIE_TAU_F_P,7,255)
C2c1:(NPDPIE_TAU_S_P,2,100)
C2c1:(NPDPIE_THR_F_P,5,200)
C2c1:(NPDPIE_THR_S_P,5,200)
C2c1:(NPDPII_TAU_F_P,7,255)
C2c1:(NPDPII_TAU_S_P,7,255)
C2c1:(NPDPII_THR_F_P,5,200)
C2c1:(NPDPII_THR_S_P,5,200)
C2c1:(PS_WEIGHT_EXC_F_N,6,200)
C2c1:(PS_WEIGHT_EXC_S_N,7,130)
C2c1:(PS_WEIGHT_INH_F_N,6,120)
C2c1:(PS_WEIGHT_INH_S_N,7,255)
C2c1:(PULSE_PWLK_P,4,106)
C2c1:(R2R_P,3,85)
C2c2:(IF_AHTAU_N,4,80)
C2c2:(IF_AHTHR_N,4,80)
C2c2:(IF_AHW_P,0,0)
C2c2:(IF_BUF_P,4,80)
C2c2:(IF_CASC_N,0,0)
C2c2:(IF_DC_P,0,0)
C2c2:(IF_NMDA_N,0,0)
C2c2:(IF_RFR_N,2,128)
C2c2:(IF_TAU1_N,3,150)
C2c2:(IF_TAU2_N,3,150)
C2c2:(IF_THR_N,0,0)
C2c2:(NPDPIE_TAU_F_P,7,255)
C2c2:(NPDPIE_TAU_S_P,2,100)
C2c2:(NPDPIE_THR_F_P,5,200)
C2c2:(NPDPIE_THR_S_P,5,200)
C2c2:(NPDPII_TAU_F_P,3,150)
C2c2:(NPDPII_TAU_S_P,3,150)
C2c2:(NPDPII_THR_F_P,5,200)
C2c2:(NPDPII_THR_S_P,5,200)
C2c2:(PS_WEIGHT_EXC_F_N,6,200)
C2c2:(PS_WEIGHT_EXC_S_N,4,50)
C2c2:(PS_WEIGHT_INH_F_N,6,120)
C2c2:(PS_WEIGHT_INH_S_N,7,255)
C2c2:(PULSE_PWLK_P,4,106)
C2c2:(R2R_P,3,85)
C2c3:(IF_AHTAU_N,4,80)
C2c3:(IF_AHTHR_N,4,80)
C2c3:(IF_AHW_P,0,0)
C2c3:(IF_BUF_P,4,80)
C2c3:(IF_CASC_N,0,0)
C2c3:(IF_DC_P,0,0)
C2c3:(IF_NMDA_N,0,0)
C2c3:(IF_RFR_N,2,128)
C2c3:(IF_TAU1_N,3,150)
C2c3:(IF_TAU2_N,3,150)
C2c3:(IF_THR_N,0,0)
C2c3:(NPDPIE_TAU_F_P,7,255)
C2c3:(NPDPIE_TAU_S_P,2,100)
C2c3:(NPDPIE_THR_F_P,5,200)
C2c3:(NPDPIE_THR_S_P,5,200)
C2c3:(NPDPII_TAU_F_P,3,150)
C2c3:(NPDPII_TAU_S_P,3,150)
C2c3:(NPDPII_THR_F_P,5,200)
C2c3:(NPDPII_THR_S_P,5,200)
C2c3:(PS_WEIGHT_EXC_F_N,6,200)
C2c3:(PS_WEIGHT_EXC_S_N,4,50)
C2c3:(PS_WEIGHT_INH_F_N,6,120)
C2c3:(PS_WEIGHT_INH_S_N,7,255)
C2c3:(PULSE_PWLK_P,4,106)
C2c3:(R2R_P,3,85)
C3c0:(IF_AHTAU_N,4,80)
C3c0:(IF_AHTHR_N,4,80)
C3c0:(IF_AHW_P,0,0)
C3c0:(IF_BUF_P,4,80)
C3c0:(IF_CASC_N,0,0)
C3c0:(IF_DC_P,0,0)
C3c0:(IF_NMDA_N,0,0)
C3c0:(IF_RFR_N,2,128)
C3c0:(IF_TAU1_N,3,150)
C3c0:(IF_TAU2_N,3,150)
C3c0:(IF_THR_N,0,0)
C3c0:(NPDPIE_TAU_F_P,7,255)
C3c0:(NPDPIE_TAU_S_P,2,100)
C3c0:(NPDPIE_THR_F_P,5,200)
C3c0:(NPDPIE_THR_S_P,5,200)
C3c0:(NPDPII_TAU_F_P,7,255)
C3c0:(NPDPII_TAU_S_P,7,255)
C3c0:(NPDPII_THR_F_P,5,200)
C3c0:(NPDPII_THR_S_P,5,200)
C3c0:(PS_WEIGHT_EXC_F_N,6,200)
C3c0:(PS_WEIGHT_EXC_S_N,7,130)
C3c0:(PS_WEIGHT_INH_F_N,6,120)
C3c0:(PS_WEIGHT_INH_S_N,7,255)
C3c0:(PULSE_PWLK_P,4,106)
C3c0:(R2R_P,3,85)
C3c1:(IF_AHTAU_N,4,80)
C3c1:(IF_AHTHR_N,4,80)
C3c1:(IF_AHW_P,0,0)
C3c1:(IF_BUF_P,4,80)
C3c1:(IF_CASC_N,0,0)
C3c1:(IF_DC_P,0,0)
C3c1:(IF_NMDA_N,0,0)
C3c1:(IF_RFR_N,2,128)
C3c1:(IF_TAU1_N,3,150)
C3c1:(IF_TAU2_N,3,150)
C3c1:(IF_THR_N,0,0)
C3c1:(NPDPIE_TAU_F_P,7,255)
C3c1:(NPDPIE_TAU_S_P,2,100)
C3c1:(NPDPIE_THR_F_P,5,200)
C3c1:(NPDPIE_THR_S_P,5,200)
C3c1:(NPDPII_TAU_F_P,7,255)
C3c1:(NPDPII_TAU_S_P,7,255)
C3c1:(NPDPII_THR_F_P,5,200)
C3c1:(NPDPII_THR_S_P,5,200)
C3c1:(PS_WEIGHT_EXC_F_N,6,200)
C3c1:(PS_WEIGHT_EXC_S_N,7,130)
C3c1:(PS_WEIGHT_INH_F_N,6,120)
C3c1:(PS_WEIGHT_INH_S_N,7,255)
C3c1:(PULSE_PWLK_P,4,106)
C3c1:(R2R_P,3,85)
C3c2:(IF_AHTAU_N,4,80)
C3c2:(IF_AHTHR_N,4,80)
C3c2:(IF_AHW_P,0,0)
C3c2:(IF_BUF_P,4,80)
C3c2:(IF_CASC_N,0,0)
C3c2:(IF_DC_P,0,0)
C3c2:(IF_NMDA_N,0,0)
C3c2:(IF_RFR_N,2,128)
C3c2:(IF_TAU1_N,3,150)
C3c2:(IF_TAU2_N,3,150)
C3c2:(IF_THR_N,0,0)
C3c2:(NPDPIE_TAU_F_P,7,255)
C3c2:(NPDPIE_TAU_S_P,2,100)
C3c2:(NPDPIE_THR_F_P,5,200)
C3c2:(NPDPIE_THR_S_P,5,200)
C3c2:(NPDPII_TAU_F_P,3,150)
C3c2:(NPDPII_TAU_S_P,3,150)
C3c2:(NPDPII_THR_F_P,5,200)
C3c2:(NPDPII_THR_S_P,5,200)
C3c2:(PS_WEIGHT_EXC_F_N,6,200)
C3c2:(PS_WEIGHT_EXC_S_N,4,50)
C3c2:(PS_WEIGHT_INH_F_N,6,120)
C3c2:(PS_WEIGHT_INH_S_N,7,255)
C3c2:(PULSE_PWLK_P,4,106)
C3c2:(R2R_P,3,85)
C3c3:(IF_AHTAU_N,4,80)
C3c3:(IF_AHTHR_N,4,80)
C3c3:(IF_AHW_P,0,0)
C3c3:(IF_BUF_P,4,80)
C3c3:(IF_CASC_N,0,0)
C3c3:(IF_DC_P,0,0)
C3c3:(IF_NMDA_N,0,0)
C3c3:(IF_RFR_N,2,128)
C3c3:(IF_TAU1_N,3,150)
C3c3:(IF_TAU2_N,3,150)
C3c3:(IF_THR_N,0,0)
C3c3:(NPDPIE_TAU_F_P,7,255)
C3c3:(NPDPIE_TAU_S_P,2,100)
C3c3:(NPDPIE_THR_F_P,5,200)
C3c3:(NPDPIE_THR_S_P,5,200)
C3c3:(NPDPII_TAU_F_P,3,150)
C3c3:(NPDPII_TAU_S_P,3,150)
C3c3:(NPDPII_THR_F_P,5,200)
C3c3:(NPDPII_THR_S_P,5,200)
C3c3:(PS_WEIGHT_EXC_F_N,6,200)
C3c3:(PS_WEIGHT_EXC_S_N,4,50)
C3c3:(PS_WEIGHT_INH_F_N,6,120)
C3c3:(PS_WEIGHT_INH_S_N,7,255)
C3c3:(PULSE_PWLK_P,4,106)
C3c3:(R2R_P,3,85)


src/params.py
import samna.dynapse1 as dyn1

def agent_param_group():
    """Generate a Dynapse1ParameterGroup of one core with some synapse
    weights turned on for examples.

    Returns:
        samna.dynapse1.Dynapse1ParameterGroup: Dynapse1ParameterGroup.
    """
    param_group = dyn1.Dynapse1ParameterGroup()
    # Neuron V_mem gain (not the firing threshold!) #chip 07
    param_group.param_map["IF_THR_N"].coarse_value = 0 # lower than 2-3
    param_group.param_map["IF_THR_N"].fine_value = 0
    # # refactory period
    param_group.param_map["IF_RFR_N"].coarse_value = 2 # the lower the bigger the refactory period
    param_group.param_map["IF_RFR_N"].fine_value = 128
    # # Main neuron time constant (leakage current)
    param_group.param_map["IF_TAU1_N"].coarse_value = 3 # higher than 2-3
    param_group.param_map["IF_TAU1_N"].fine_value = 150
    # # Secondary neuron time constant
    param_group.param_map["IF_TAU2_N"].coarse_value = 3
    param_group.param_map["IF_TAU2_N"].fine_value = 150
    # # DC current injected into a neuron
    param_group.param_map["IF_DC_P"].coarse_value = 0
    param_group.param_map["IF_DC_P"].fine_value = 0
    
    ### AMPA snyapse ###
    # time constant (leakage)
    param_group.param_map["NPDPIE_TAU_F_P"].coarse_value = 7
    param_group.param_map["NPDPIE_TAU_F_P"].fine_value = 255
    # synapses threshold (gain)
    param_group.param_map["NPDPIE_THR_F_P"].coarse_value = 5
    param_group.param_map["NPDPIE_THR_F_P"].fine_value = 200
    # synapse weight
    param_group.param_map["PS_WEIGHT_EXC_F_N"].coarse_value = 6
    param_group.param_map["PS_WEIGHT_EXC_F_N"].fine_value = 200
    
    ### NMDA snyapse ###
    # time constant (leakage)
    param_group.param_map["NPDPIE_TAU_S_P"].coarse_value = 2
    param_group.param_map["NPDPIE_TAU_S_P"].fine_value = 100
    # # synapses threshold
    param_group.param_map["NPDPIE_THR_S_P"].coarse_value = 5
    param_group.param_map["NPDPIE_THR_S_P"].fine_value = 200
    # # synapses weight
    param_group.param_map["PS_WEIGHT_EXC_S_N"].coarse_value = 7
    param_group.param_map["PS_WEIGHT_EXC_S_N"].fine_value = 130

    ### GABA_A snyapse ###
    # time constant (leakage)
    param_group.param_map["NPDPII_TAU_F_P"].coarse_value = 7
    param_group.param_map["NPDPII_TAU_F_P"].fine_value = 255
    # # synapses threshold
    param_group.param_map["NPDPII_THR_F_P"].coarse_value = 5
    param_group.param_map["NPDPII_THR_F_P"].fine_value = 200
    # # synapse weights
    param_group.param_map["PS_WEIGHT_INH_F_N"].coarse_value = 6
    param_group.param_map["PS_WEIGHT_INH_F_N"].fine_value = 120

    
    # ### GABA_B snyapse ###
    # # time constant (leakage)
    param_group.param_map["NPDPII_TAU_S_P"].coarse_value = 7
    param_group.param_map["NPDPII_TAU_S_P"].fine_value = 255
    # # synapses threshold
    param_group.param_map["NPDPII_THR_S_P"].coarse_value = 5
    param_group.param_map["NPDPII_THR_S_P"].fine_value = 200
    # # synapse weights
    param_group.param_map["PS_WEIGHT_INH_S_N"].coarse_value = 7
    param_group.param_map["PS_WEIGHT_INH_S_N"].fine_value = 255

    return param_group

def model_param_group():
    """Generate a Dynapse1ParameterGroup of one core with some synapse
    weights turned on for examples.

    Returns:
        samna.dynapse1.Dynapse1ParameterGroup: Dynapse1ParameterGroup.
    """
    param_group = dyn1.Dynapse1ParameterGroup()
    # Neuron V_mem gain (not the firing threshold!) 
    param_group.param_map["IF_THR_N"].coarse_value = 0 # lower than 2-3
    param_group.param_map["IF_THR_N"].fine_value = 0
    # # refactory period
    param_group.param_map["IF_RFR_N"].coarse_value = 2 # the lower the bigger the refactory period
    param_group.param_map["IF_RFR_N"].fine_value = 128
    # # Main neuron time constant (leakage current)
    param_group.param_map["IF_TAU1_N"].coarse_value = 3 # higher than 2-3
    param_group.param_map["IF_TAU1_N"].fine_value = 150
    # # Secondary neuron time constant
    param_group.param_map["IF_TAU2_N"].coarse_value = 3
    param_group.param_map["IF_TAU2_N"].fine_value = 150
    # # DC current injected into a neuron
    param_group.param_map["IF_DC_P"].coarse_value = 0
    param_group.param_map["IF_DC_P"].fine_value = 0
    
    ### AMPA snyapse ###
    # time constant (leakage)
    param_group.param_map["NPDPIE_TAU_F_P"].coarse_value = 7
    param_group.param_map["NPDPIE_TAU_F_P"].fine_value = 255
    # synapses threshold (gain)
    param_group.param_map["NPDPIE_THR_F_P"].coarse_value = 5
    param_group.param_map["NPDPIE_THR_F_P"].fine_value = 200
    # synapse weight
    param_group.param_map["PS_WEIGHT_EXC_F_N"].coarse_value = 6
    param_group.param_map["PS_WEIGHT_EXC_F_N"].fine_value = 200
    
    ### NMDA snyapse ###
    # time constant (leakage)
    param_group.param_map["NPDPIE_TAU_S_P"].coarse_value = 2
    param_group.param_map["NPDPIE_TAU_S_P"].fine_value = 100
    # # synapses threshold
    param_group.param_map["NPDPIE_THR_S_P"].coarse_value = 5
    param_group.param_map["NPDPIE_THR_S_P"].fine_value = 200
    # # synapses weight
    param_group.param_map["PS_WEIGHT_EXC_S_N"].coarse_value = 4
    param_group.param_map["PS_WEIGHT_EXC_S_N"].fine_value = 50

    ### GABA_A snyapse ###
    # time constant (leakage)
    param_group.param_map["NPDPII_TAU_F_P"].coarse_value = 3
    param_group.param_map["NPDPII_TAU_F_P"].fine_value = 150
    # # synapses threshold
    param_group.param_map["NPDPII_THR_F_P"].coarse_value = 5
    param_group.param_map["NPDPII_THR_F_P"].fine_value = 200
    # # synapse weights
    param_group.param_map["PS_WEIGHT_INH_F_N"].coarse_value = 6
    param_group.param_map["PS_WEIGHT_INH_F_N"].fine_value = 120

    # ### GABA_B snyapse ###
    # # time constant (leakage)
    param_group.param_map["NPDPII_TAU_S_P"].coarse_value = 3
    param_group.param_map["NPDPII_TAU_S_P"].fine_value = 150
    # # synapses threshold
    param_group.param_map["NPDPII_THR_S_P"].coarse_value = 5
    param_group.param_map["NPDPII_THR_S_P"].fine_value = 200
    # # synapse weights
    param_group.param_map["PS_WEIGHT_INH_S_N"].coarse_value = 7
    param_group.param_map["PS_WEIGHT_INH_S_N"].fine_value = 255


    return param_group